xfrd-tcp.c

```diff
@@ -24,6 +24,82 @@
 #include "xfrd.h"
 #include "xfrd-disk.h"
 #include "util.h"
+#ifdef HAVE_TLS_1_3
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#endif
+
+#ifdef HAVE_TLS_1_3
+static SSL_CTX*
+create_ssl_context()
+{
+	SSL_CTX *ctx;
+	ctx = SSL_CTX_new(TLS_client_method());
+	if (!ctx) {
+		log_msg(LOG_ERR, "xfrd tls: Unable to create SSL ctxt");
+	}
+	else if (SSL_CTX_set_default_verify_paths(ctx) != 1) {
+		SSL_CTX_free(ctx);
+		log_msg(LOG_ERR, "xfrd tls: Unable to set default SSL verify paths");
+	}
```

Comment:
We should not return the deallocated ctx pointer, to stop double free errors.

```suggestion
	else if (SSL_CTX_set_default_verify_paths(ctx) != 1) {
		SSL_CTX_free(ctx);
		log_msg(LOG_ERR, "xfrd tls: Unable to set default SSL verify paths");
		return NULL;
	}
```

Reply:
This has been resolved in the current code.